### PR TITLE
[e2e-test] Add e2e_bootup_bad_rom_ext_signature test for CW310

### PR DESCRIPTION
**Pull request by milesdai**
Includes a number of autogenerated files from `cargo raze` unfortunately.
Also, a number of files were copied over from `opentitantool/src/command` to bring in some basic bootstrapping and bitstream loading functionality.

### CODE REVIEW

**Comment by cfrantz** [comment]
Rather than commenting on the code, I thought I'd to a bit of re-arranging and a single explanation:

As we discussed yesterday, I want the E2E tests to be stand alone executables that make use of opentitanlib to accomplish their testing goals.  Most tests have some amount of common initialization:

1. Initializing the Transport
2. Loading a bitstream
3. Loading a RV32 test program (e.g. bootstrap).

I've refactored all of that into a `helpers` module.  Really, this should be its own library or module within opentitanlib.  Notably, the helpers module exports a struct `InitializeTest` that does all of this initialization and returns a `Transport`.  The `InitializeTest` struct contains all of the basic/common test command line flags, aggregated together using `StructOpt`s "flatten" feature (run bad_rom_ext_signature --help` to see).

Every test program can reuse the common initialization sequence (see the `main` function).

I've refactored the actual test itself into its own function and initialized a `UartConsole` struct to look for the
pass/fail regexes.  The console doesn't accept `stdin` since we don't need it; we do give it a `stdout` as a courtesy to anyone looking at the test logs.

**Review by timothytrippel — APPROVED** [review]
Really great, to see this come together, nice job @milesdai ! 

As mentioned, I will have to update the airgapped image before merging since the rust deps have changed, or you can give it a shot if you want and I can walk you through it.

File: sw/device/silicon_creator/mask_rom/BUILD [hunk 1]
```diff
@@ -347,3 +347,27 @@ opentitan_functest(
...
+        args = [
+            "--rom-ext=\"$(location {flash})\"",
+            "--rom-kind=maskrom",
+            "--bitstream=\"$(location //hw/bitstream:mask_rom)\"",
+        ],
+        data = [
+            "//hw/bitstream:mask_rom",
+            "//sw/host/tests/mask_rom/e2e_bootup_bad_rom_ext_signature:test_resources",
+        ],
+        test_runner = "//util:opentitan_functest_runner.sh",
```
Comment (cfrantz) [inline sw/device/silicon_creator/mask_rom/BUILD:365]: We shouldn't need this, correct?  (its already part of the BASE_PARAMS)

File: sw/host/tests/mask_rom/e2e_bootup_bad_rom_ext_signature/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+
+    #[structopt(
+        long, parse(try_from_str=humantime::parse_duration),
+        default_value = "60s",
+        help = "Duration of ROM detection timeout",
```
Comment (cfrantz) [inline sw/host/tests/mask_rom/e2e_bootup_bad_rom_ext_signature/src/main.rs:26]: I think this help text is wrong.  This is really the rom_ext boot detection timeout, correct?
I also think this timeout is probably way too long, at least for cw310.  I think this is probably more like 5s. (In reality the rom_ext boots in less than a second)

**Review by cfrantz — COMMENTED** [review]
Just a couple of nits.

File: sw/device/silicon_creator/mask_rom/BUILD (hunk 1 above)
Comment (milesdai) [reply sw/device/silicon_creator/mask_rom/BUILD:365]: Fixed! I forgot I had replaced opentitantool_test_runner.sh.

File: sw/host/tests/mask_rom/e2e_bootup_bad_rom_ext_signature/src/main.rs (hunk 2 above)
Comment (milesdai) [reply sw/host/tests/mask_rom/e2e_bootup_bad_rom_ext_signature/src/main.rs:26]: I've updated the help text to `ROM_EXT boot detection timeout` and changed the default to 5s.

**Review by cfrantz — APPROVED** [review]

**Pull request merged**